src/tasks/LargeFileUploadTask.ts

```diff
@@ -0,0 +1,365 @@
+ */
+export interface UploadSession {
+  expirationDateTime?: Date | null;
+  nextExpectedRanges?: string[] | null;
+  odataType?: string | null;
+  uploadUrl?: string | null;
+}
+
+/**
+ * @interface
+ * Signature to represent the result of an upload
+ */
+export interface UploadResult<T> {
+  itemResponse?: T | null;
+  location?: string;
+}
+
+/**
+ * UploadSession ParsableFactory
+ * Creates a factory function to deserialize the upload session response.
+ *
+ * @param {ParseNode} _parseNode - The parse node to deserialize.
+ * @returns {Function} - A function that takes an instance of Parsable and returns a record of deserialization functions.
+ */
+export const createUploadSessionFromDiscriminatorValue = (
+  _parseNode: ParseNode | undefined,
+): ((instance?: Parsable) => Record<string, (node: ParseNode) => void>) => {
+  return deserializeIntoUploadSession;
+};
+
+/**
+ * Deserializes the upload session response body.
+ *
+ * @param {Partial<UploadSession>} [uploadSession] - The upload session object to deserialize into.
+ * @returns {Record<string, (node: ParseNode) => void>} - A record of deserialization functions.
+ */
+export const deserializeIntoUploadSession = (
+  uploadSession: Partial<UploadSession> | undefined = {},
+): Record<string, (node: ParseNode) => void> => {
+  return {
+    expirationDateTime: n => {
+      uploadSession.expirationDateTime = n.getDateValue();
+    },
+    nextExpectedRanges: n => {
+      uploadSession.nextExpectedRanges = n.getCollectionOfPrimitiveValues();
+    },
+    "@odata.type": n => {
+      uploadSession.odataType = n.getStringValue();
+    },
+    uploadUrl: n => {
+      uploadSession.uploadUrl = n.getStringValue();
+    },
+  };
+};
+
+/**
+ * @constant
+ * A default slice size for a large file
+ */
+const DefaultSliceSize = 320 * 1024;
+
+/**
+ * A class representing LargeFileUploadTask
+ */
+export class LargeFileUploadTask<T extends Parsable> {
+  /**
+   * @private
+   * The ranges to upload
+   */
+  rangesRemaining: number[][] = [];
+
+  /**
+   * @private
+   * The error mappings
+   */
+  errorMappings: ErrorMappings;
+
+  /**
+   * @private
+   * The upload session
+   */
+  Session: UploadSession;
+
+  /**
+   * Constructs a new instance of the LargeFileUploadTask class.
+   *
+   * @param {RequestAdapter} requestAdapter - The request adapter to use for making HTTP requests.
+   * @param {Parsable} uploadSession - The upload session information.
+   * @param {ReadableStream<Uint8Array>} uploadStreamProvider - Returns an instance of an unconsumed new stream to be uploaded.
+   * @param {number} [maxSliceSize=-1] - The maximum size of each file slice to be uploaded.
+   * @param {ParsableFactory<T>} parsableFactory - The factory to create parsable objects.
+   * @param {ErrorMappings} [errorMappings] - error mappings.
+   *
+   * @throws {Error} If any of the required parameters are undefined or invalid.
+   */
+  constructor(
+    readonly requestAdapter: RequestAdapter,
```

Comment:
is the default visibility of this syntax private? should we add the private keyword here?